presto-spi/src/main/java/com/facebook/presto/spi/plan/MergeJoinNode.java

```diff
@@ -82,6 +83,15 @@ public MergeJoinNode(
         this.rightHashVariable = requireNonNull(rightHashVariable, "rightHashVariable is null");
     }
 
+    /**
+     * @param joinNode
+     * @return returns true if merge join is supported for the given join node.
+     */
+    public static boolean isMergeJoinEligible(JoinNode joinNode)
```

Comment:
Perhaps moving the util function to the optimizer itself. It's we support inner join for now onlyl